gadgets/trace_open/test/unit/trace_open_test.go

```diff
@@ -0,0 +1,297 @@
+						Pid:   info.Pid,
+						Tid:   info.Tid,
+						Uid:   uint32(info.Uid),
+						Gid:   uint32(info.Gid),
+						Fd:    uint32(fd),
+						FName: "/tmp/test_symbolic_links",
+					}
+				})(t, info, fd, events)
+				return nil
+			},
+		},
+		"test_relative_path": {
+			runnerConfig: &utilstest.RunnerConfig{},
+			generateEvent: func(t *testing.T) int {
+				relPath := generateRelativePathForAbsolutePath(t, "/tmp/test_relative_path")
+				fd, err := unix.Open(relPath, unix.O_CREAT|unix.O_RDWR, unix.S_IRWXU|unix.S_IRGRP|unix.S_IWGRP|unix.S_IXOTH)
+				require.NoError(t, err, "opening file")
+
+				defer os.Remove(relPath)
+
+				unix.Close(fd)
+
+				return fd
+			},
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, fd int, events []ExpectedTraceOpenEvent) error {
+				utilstest.ExpectAtLeastOneEvent(func(info *utilstest.RunnerInfo, fd int) *ExpectedTraceOpenEvent {
+					return &ExpectedTraceOpenEvent{
+						Comm:  info.Comm,
+						Pid:   info.Pid,
+						Tid:   info.Tid,
+						Uid:   uint32(info.Uid),
+						Gid:   uint32(info.Gid),
+						Fd:    uint32(fd),
+						FName: generateRelativePathForAbsolutePath(t, "/tmp/test_relative_path"),
+					}
+				})(t, info, fd, events)
+				return nil
+			},
+		},
+		"test_prefix_on_directory": {
+			runnerConfig: &utilstest.RunnerConfig{},
+			generateEvent: func(t *testing.T) int {
+				err := os.Mkdir("/tmp/foo", 0o750)
+				if err != nil {
+					require.NoError(t, err, "mkdir for /tmp/foo")
+					return 0
+				}
+				defer os.RemoveAll("/tmp/foo")
+
+				fd, err := unix.Open("/tmp/foo/bar.test", unix.O_RDONLY|unix.O_CREAT, 0)
+				if err != nil {
+					require.NoError(t, err, "opening /tmp/foo")
+					return 0
+				}
+				defer unix.Close(fd)
+
+				badfd, err := unix.Open("/tmp/quux.test", unix.O_RDONLY|unix.O_CREAT, 0)
+				if err != nil {
+					require.NoError(t, err, "opening /tmp/quux.test")
+					return 0
+				}
+				defer unix.Close(badfd)
+
+				return fd
+			},
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, fd int, events []ExpectedTraceOpenEvent) error {
+				utilstest.ExpectAtLeastOneEvent(func(info *utilstest.RunnerInfo, fd int) *ExpectedTraceOpenEvent {
+					return &ExpectedTraceOpenEvent{
+						Comm:  info.Comm,
+						Pid:   info.Pid,
+						Tid:   info.Tid,
+						Uid:   uint32(info.Uid),
+						Gid:   uint32(info.Gid),
+						Fd:    uint32(fd),
+						FName: "/tmp/foo/bar.test",
+					}
+				})(t, info, fd, events)
+				return nil
+			}},
+	}
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			timeout := 5 * time.Second
+			var fd int
+			runner := utilstest.NewRunnerWithTest(t, testCase.runnerConfig)
+			apiParams := map[string]string{
+				"operator.oci.ebpf.uid": "0",
+			}
+			var mntsFilterMap *ebpf.Map
+			if testCase.mntsFilterMap != nil {
+				mntsFilterMap = testCase.mntsFilterMap(runner.Info)
+			}
+			gdgt, err := gadgetapi.NewGadget(
+				"trace_open",
+				"latest",
```

Comment:
Please make this configurable. You can take some inspiration from pkg/testing/ig/ig.go.